kyuubi-server/src/test/scala/org/apache/kyuubi/session/KyuubiBatchSessionSuite.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.session
+
+import org.apache.kyuubi.{Utils, WithKyuubiServer}
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.engine.spark.SparkBatchProcessBuilder
+import org.apache.kyuubi.server.api.v1.{BatchesResource, BatchRequest}
+
+class KyuubiBatchSessionSuite extends WithKyuubiServer {
```

Comment:
use e2e tests in BatchesResourceSuite?

Reply:
this test is for checking the internal config propagation between request -> session -> builder. e2e test can not catch the details of each class, so the unit test should be good ?

Reply:
so why do we need a WithKyuubiServer injection?

Reply:
for example, the tests you added behave complete different in batch or SQL sessions. For SQL, k1/k2 will be sent to SparkSQLEngine with the opening session, but for batch, there are spark.k1/spark.v2 only

Reply:
Without `WithKyuubiServer`, the event framwork will fail since the kyuubi server is not initialized..

I added some more test for checking the commands result.